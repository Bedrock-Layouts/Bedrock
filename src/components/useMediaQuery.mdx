---
name: useMediaQuery
menu: Components
---

# useMediaQuery

## Use case

When you need to do logic based on the size of the viewport, such as determining whether to use a mobile layout or full screen layout. It uses the matchMedia API, taking a value and returning a boolean for whether the viewport size is less than or equal to that value.

## How to install

```bash
  npm install @bedrock-layout/use-media-query
```

## Basic usage

```javascript
import useMediaQuery from '@bedrock-layout/use-media-query';

const App = () => {
  const smallScreen = useMediaQuery("(max-width:640px)")

  return (
    <div ref={setRef} {...props}>
      {smallScreen ? <MobileNav /> : <FullScreenNav />}
    </div>
  )
}
```

## Storybook

[Checkout storybook](https://storybook.bedrock-layout.dev/?path=/story/usemediaquery--basic) for more details.
