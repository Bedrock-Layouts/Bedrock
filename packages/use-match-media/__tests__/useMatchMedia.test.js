import useMediaQuery from "@bedrock-layout/use-media-query";
import React from "react";
import ReactDOM from "react-dom/";
import { act } from "react-dom/test-utils";

import useMatchMedia from "../src";

jest.mock("@bedrock-layout/use-media-query");

useMediaQuery.mockImplementation(() => true);

let matches;
const HookWrapper = () => {
  matches = useMatchMedia();

  return null;
};

describe("useMatchMedia", () => {
  let container;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    document.body.removeChild(container);
    container = null;
  });
  test("useMatchMedia is not null", () => {
    expect(useMatchMedia).toBeTruthy();
  });

  it("should call useMediaQuery", () => {
    act(() => {
      ReactDOM.render(<HookWrapper />, container);
    });

    expect(useMediaQuery).toBeCalled();

    expect(matches).toMatchSnapshot();
    expect(useMediaQuery.mock.calls).toMatchSnapshot();
  });
});
