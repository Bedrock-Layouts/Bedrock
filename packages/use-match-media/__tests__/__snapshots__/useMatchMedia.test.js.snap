// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useMatchMedia should call useMediaQuery 1`] = `
Object {
  "large": true,
  "largeOnly": true,
  "medium": true,
  "mediumOnly": true,
  "smallOnly": true,
  "xlarge": true,
  "xlargeOnly": true,
  "xxlarge": true,
}
`;

exports[`useMatchMedia should call useMediaQuery 2`] = `
Array [
  Array [
    "(max-width:639px)",
  ],
  Array [
    "(min-width:640px) and (max-width:1023px)",
  ],
  Array [
    "(min-width:1024px) and (max-width:1199px)",
  ],
  Array [
    "(min-width:1200px) and (max-width:1439px)",
  ],
  Array [
    "(max-width:1023px)",
  ],
  Array [
    "(max-width:1199px)",
  ],
  Array [
    "(max-width:1439px)",
  ],
  Array [
    "(max-width:1920px)",
  ],
]
`;
